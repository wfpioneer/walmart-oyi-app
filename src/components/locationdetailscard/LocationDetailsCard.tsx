import React from 'react';
import {
  GestureResponderEvent,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import styles from './locationDetailsCard.style';
import { COLOR } from '../../themes/Color';

export interface Props {
  locationName: string;
  locationType: string;
  editAction?(event: GestureResponderEvent): void;
  deleteAction?(event: GestureResponderEvent): void;
}

function locationDetailsCard(props: Props) {
  const renderIcon = (name: string, action?: (event: GestureResponderEvent) => void) => (
    <TouchableOpacity testID={name} style={styles.icon} onPress={action}>
      <View>
        <MaterialCommunityIcon name={name} size={20} color={COLOR.TRACKER_GREY} />
      </View>
    </TouchableOpacity>
  );

  return (
    <View style={styles.card}>
      <View style={styles.location}><Text style={styles.locationName}>{props.locationName}</Text></View>
      {props.locationType !== 'Reserve' && renderIcon('pencil', props.editAction)}
      {props.locationType !== 'Reserve' && renderIcon('trash-can', props.deleteAction)}
    </View>
  );
}

export default locationDetailsCard;
