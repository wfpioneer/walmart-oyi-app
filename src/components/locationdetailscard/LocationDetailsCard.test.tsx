import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import LocationDetailsCard from './LocationDetailsCard';

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'mockMaterialCommunityIcons');

describe('Tests rendering LocationDetailsCard', () => {
  it('Renders the LocationDetailsCard for location type reserve', () => {
    const { toJSON } = render(
      <LocationDetailsCard
        locationName="A1-1"
        locationType="Reserve"
        editAction={jest.fn()}
        deleteAction={jest.fn()}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('Renders the LocationDetailsCard for location type floor and onaction btn click call mock function', () => {
    const mockEdit = jest.fn();
    const mockDelete = jest.fn();

    const { getByTestId, toJSON } = render(
      <LocationDetailsCard
        locationName="A1-1"
        locationType="Floor"
        editAction={mockEdit}
        deleteAction={mockDelete}
      />
    );
    expect(toJSON()).toMatchSnapshot();
    const editBtn = getByTestId('pencil');
    const deleteBtn = getByTestId('trash-can');
    fireEvent.press(editBtn);
    fireEvent.press(deleteBtn);
    expect(mockEdit).toBeCalledTimes(1);
    expect(mockDelete).toBeCalledTimes(1);
  });
});
