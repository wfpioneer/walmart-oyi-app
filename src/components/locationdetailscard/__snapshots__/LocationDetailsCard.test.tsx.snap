// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests rendering LocationDetailsCard Renders the LocationDetailsCard for location type floor and onaction btn click call mock function 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "height": 70,
      "paddingLeft": 10,
      "paddingRight": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
        "flex": 8,
        "paddingBottom": 15,
        "paddingTop": 15,
        "textTransform": "uppercase",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#545F7A",
          "fontWeight": "bold",
        }
      }
    >
      A1-1
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 15,
        "paddingTop": 15,
      }
    }
    testID="pencil"
  >
    <View>
      <mockMaterialCommunityIcons
        color="#545F7A"
        name="pencil"
        size={20}
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 15,
        "paddingTop": 15,
      }
    }
    testID="trash-can"
  >
    <View>
      <mockMaterialCommunityIcons
        color="#545F7A"
        name="trash-can"
        size={20}
      />
    </View>
  </View>
</View>
`;

exports[`Tests rendering LocationDetailsCard Renders the LocationDetailsCard for location type reserve 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "height": 70,
      "paddingLeft": 10,
      "paddingRight": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
        "flex": 8,
        "paddingBottom": 15,
        "paddingTop": 15,
        "textTransform": "uppercase",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#545F7A",
          "fontWeight": "bold",
        }
      }
    >
      A1-1
    </Text>
  </View>
</View>
`;
